### PR TITLE
Color vote if you have voted f3 or f4.

### PR DESCRIPTION
**Pull request by furo321**
Color the "Vote yes" or "Vote no" text depending on which you have voted on.
image
image

## Checklist

- [x] Tested the change ingame
- [x] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

**Comment by heinrich5991** [comment]
Looks cool. :)

Does this work if you press f4 after pressing f3?

**Comment by furo321** [comment]
Yes, it will change the color to what the new vote is.

**Comment by heinrich5991** [comment]
This isn't the behavior on the server though, I think. The second vote will be ignored.

Maybe this could be ignored locally? It'd be best if the server could send the current vote back to the client, so it could manipulate the vote state.

**Comment by furo321** [comment]
The second vote will only be ignored if it was sent within 3 seconds of the last vote (when sv_spamprotection is enabled, which it is by default). It would be great if the server could send back the player's current vote.

**Comment by heinrich5991** [comment]
Can you implement the server sending what the client voted, so we have correct information for this UI.

I think adding UI that is sometimes wrong is not the way to go.

**Comment by furo321** [comment]
I've added the `Sv_YourVote` packet now, it's only send to clients with a ClientVersion above or equal to 17022. If the server doesn't support the packet, then the client will only color the first vote you send.

File: src/game/server/gamecontext.cpp [hunk 1]
```diff
@@ -2371,6 +2371,12 @@ void CGameContext::OnVoteNetMessage(const CNetMsg_Cl_Vote *pMsg, int ClientID)
 	pPlayer->m_Vote = pMsg->m_Vote;
 	pPlayer->m_VotePos = ++m_VotePos;
 	m_VoteUpdate = true;
+
+	if(GetClientVersion(ClientID) >= 17022)
```
Comment (Robyt3) [inline src/game/server/gamecontext.cpp:2375]: This should be a constant, defined after this line:

https://github.com/ddnet/ddnet/blob/886962b561cdb02302db2ec5a16a36af6ae24cc2/src/engine/shared/protocol.h#L127

The next version with this feature will likely be `17030`.

File: src/game/server/gamecontext.cpp [hunk 2]
```diff
@@ -2371,6 +2371,12 @@ void CGameContext::OnVoteNetMessage(const CNetMsg_Cl_Vote *pMsg, int ClientID)
 	pPlayer->m_Vote = pMsg->m_Vote;
 	pPlayer->m_VotePos = ++m_VotePos;
 	m_VoteUpdate = true;
+
+	if(GetClientVersion(ClientID) >= VERSION_DDNET_YOURVOTE)
```
Comment (heinrich5991) [inline src/game/server/gamecontext.cpp:2375]: Actually, just drop this check. It doesn't hurt to send unknown messages to older clients.

**Review by heinrich5991 — COMMENTED** [review]
Looks good to me. :)

Please drop the version check, it doesn't add anything meaningful.

**Review by heinrich5991 — APPROVED** [review]
Thanks! I squashed your commits so we don't have intermediate states that are unnecessary.

**Pull request merged**